includes/utils.php

```diff
@@ -416,27 +416,35 @@ function get_term_tree( $all_terms, $orderby = 'count', $order = 'desc', $flat =
 	}
 
 	if ( $flat ) {
-		$flat_tree = [];
-
-		foreach ( $terms_tree as $term ) {
-			$flat_tree[] = $term;
-			$to_process  = $term->children;
-			while ( ! empty( $to_process ) ) {
-				$term        = array_shift( $to_process );
-				$flat_tree[] = $term;
-
-				if ( ! empty( $term->children ) ) {
-					$to_process = $term->children + $to_process;
-				}
-			}
-		}
+		$flat_tree =  flatten_tree_array( $terms_tree );
 
 		return $flat_tree;
```

Comment:
Please replace with the upstream fix:
https://github.com/10up/ElasticPress/commit/cd4183708c63f45a064220eb4b19866d1f417f37

So changing `$to_process = $term->children + $to_process;` to `$to_process = array_merge( $term->children, $to_process );`

The reason this was failing for your use case. is that `+` will not merge items with the same index.